<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
  <class>RecordingWidget</class>
  <widget class="QWidget" name="RecordingWidget">
    <property name="geometry">
      <rect>
        <x>0</x>
        <y>0</y>
        <width>800</width>
        <height>600</height>
      </rect>
    </property>
    <property name="windowTitle">
      <string>Recording</string>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout">
      <item>
        <widget class="QGroupBox" name="controlsGroupBox">
          <property name="title">
            <string>Recording Controls</string>
          </property>
          <layout class="QHBoxLayout" name="controlsLayout">
            <item>
              <widget class="QPushButton" name="startRecordingButton">
                <property name="text">
                  <string>Start Recording</string>
                </property>
                <property name="icon">
                  <iconset>
                    <normaloff>:/icons/record.png</normaloff>:/icons/record.png</iconset>
                </property>
                <property name="iconSize">
                  <size>
                    <width>24</width>
                    <height>24</height>
                  </size>
                </property>
              </widget>
            </item>
            <item>
              <widget class="QPushButton" name="stopRecordingButton">
                <property name="enabled">
                  <bool>false</bool>
                </property>
                <property name="text">
                  <string>Stop Recording</string>
                </property>
                <property name="icon">
                  <iconset>
                    <normaloff>:/icons/stop.png</normaloff>:/icons/stop.png</iconset>
                </property>
                <property name="iconSize">
                  <size>
                    <width>24</width>
                    <height>24</height>
                  </size>
                </property>
              </widget>
            </item>
            <item>
              <spacer name="horizontalSpacer">
                <property name="orientation">
                  <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                  <size>
                    <width>40</width>
                    <height>20</height>
                  </size>
                </property>
              </spacer>
            </item>
            <item>
              <widget class="QLabel" name="recordingTimeLabel">
                <property name="styleSheet">
                  <string>font-family: monospace; font-size: 14px; font-weight: bold;</string>
                </property>
                <property name="text">
                  <string>00:00:00</string>
                </property>
              </widget>
            </item>
          </layout>
        </widget>
      </item>
      <item>
        <widget class="QGroupBox" name="optionsGroupBox">
          <property name="title">
            <string>Recording Options</string>
          </property>
          <layout class="QGridLayout" name="optionsLayout">
            <item row="0" column="0">
              <widget class="QCheckBox" name="captureMouseCheckBox">
                <property name="text">
                  <string>Capture Mouse Events</string>
                </property>
                <property name="checked">
                  <bool>true</bool>
                </property>
              </widget>
            </item>
            <item row="0" column="1">
              <widget class="QCheckBox" name="captureKeyboardCheckBox">
                <property name="text">
                  <string>Capture Keyboard Events</string>
                </property>
                <property name="checked">
                  <bool>true</bool>
                </property>
              </widget>
            </item>
            <item row="1" column="0">
              <widget class="QCheckBox" name="optimizeMovementCheckBox">
                <property name="text">
                  <string>Optimize Mouse Movements</string>
                </property>
                <property name="checked">
                  <bool>true</bool>
                </property>
              </widget>
            </item>
            <item row="1" column="1">
              <widget class="QLabel" name="thresholdLabel">
                <property name="text">
                  <string>Movement Threshold:</string>
                </property>
              </widget>
            </item>
            <item row="1" column="2">
              <widget class="QSpinBox" name="movementThresholdSpinBox">
                <property name="suffix">
                  <string> px</string>
                </property>
                <property name="minimum">
                  <number>0</number>
                </property>
                <property name="maximum">
                  <number>100</number>
                </property>
                <property name="value">
                  <number>5</number>
                </property>
              </widget>
            </item>
          </layout>
        </widget>
      </item>
      <item>
        <widget class="QGroupBox" name="statisticsGroupBox">
          <property name="title">
            <string>Recording Statistics</string>
          </property>
          <layout class="QGridLayout" name="statisticsLayout">
            <item row="0" column="0">
              <widget class="QLabel" name="eventsCountLabel">
                <property name="text">
                  <string>Events Recorded:</string>
                </property>
              </widget>
            </item>
            <item row="0" column="1">
              <widget class="QLabel" name="eventsCountValue">
                <property name="styleSheet">
                  <string>font-weight: bold;</string>
                </property>
                <property name="text">
                  <string>0</string>
                </property>
              </widget>
            </item>
            <item row="0" column="2">
              <widget class="QLabel" name="mouseEventsLabel">
                <property name="text">
                  <string>Mouse Events:</string>
                </property>
              </widget>
            </item>
            <item row="0" column="3">
              <widget class="QLabel" name="mouseEventsValue">
                <property name="styleSheet">
                  <string>font-weight: bold;</string>
                </property>
                <property name="text">
                  <string>0</string>
                </property>
              </widget>
            </item>
            <item row="1" column="0">
              <widget class="QLabel" name="keyboardEventsLabel">
                <property name="text">
                  <string>Keyboard Events:</string>
                </property>
              </widget>
            </item>
            <item row="1" column="1">
              <widget class="QLabel" name="keyboardEventsValue">
                <property name="styleSheet">
                  <string>font-weight: bold;</string>
                </property>
                <property name="text">
                  <string>0</string>
                </property>
              </widget>
            </item>
            <item row="1" column="2">
              <widget class="QLabel" name="recordingDurationLabel">
                <property name="text">
                  <string>Duration:</string>
                </property>
              </widget>
            </item>
            <item row="1" column="3">
              <widget class="QLabel" name="recordingDurationValue">
                <property name="styleSheet">
                  <string>font-weight: bold;</string>
                </property>
                <property name="text">
                  <string>00:00:00</string>
                </property>
              </widget>
            </item>
          </layout>
        </widget>
      </item>
      <item>
        <widget class="QGroupBox" name="eventsGroupBox">
          <property name="title">
            <string>Recorded Events</string>
          </property>
          <layout class="QVBoxLayout" name="eventsLayout">
            <item>
              <widget class="QTableWidget" name="eventsTableWidget">
                <property name="alternatingRowColors">
                  <bool>true</bool>
                </property>
                <property name="selectionBehavior">
                  <enum>QAbstractItemView::SelectRows</enum>
                </property>
                <property name="sortingEnabled">
                  <bool>true</bool>
                </property>
                <column>
                  <property name="text">
                    <string>Time</string>
                  </property>
                </column>
                <column>
                  <property name="text">
                    <string>Type</string>
                  </property>
                </column>
                <column>
                  <property name="text">
                    <string>Details</string>
                  </property>
                </column>
              </widget>
            </item>
            <item>
              <layout class="QHBoxLayout" name="eventsButtonsLayout">
                <item>
                  <widget class="QPushButton" name="clearEventsButton">
                    <property name="text">
                      <string>Clear Events</string>
                    </property>
                  </widget>
                </item>
                <item>
                  <spacer name="eventsButtonsSpacer">
                    <property name="orientation">
                      <enum>Qt::Horizontal</enum>
                    </property>
                    <property name="sizeHint" stdset="0">
                      <size>
                        <width>40</width>
                        <height>20</height>
                      </size>
                    </property>
                  </spacer>
                </item>
                <item>
                  <widget class="QPushButton" name="exportEventsButton">
                    <property name="text">
                      <string>Export Events...</string>
                    </property>
                  </widget>
                </item>
              </layout>
            </item>
          </layout>
        </widget>
      </item>
    </layout>
  </widget>
  <resources/>
  <connections/>
</ui>
